### ISSUE TITLE
kube-register is broken with 0.14.x RC

### ISSUE THREAD

**Issue by AntonioMeireles** [GoogleCloudPlatform/kubernetes#6212]
upstream bug is kelseyhightower/kube-register#13. basically this breaks most of the examples in docs/source tree, as minions aren't able to register. as soon as this gets handled all of those will have to point to the new URL of the fixed version...

[filling a bug here too, as a placeholder, in order to make it easier to track for people hitting it]

**Comment by pires** [comment]
This one is heavy alright.

**Comment by lavalamp** [comment]
@brendanburns @bgrant0607 Is this a component we want to support natively, and if so, post 1.0 or pre 1.0?

**Comment by bgrant0607** [comment]
@lavalamp A breaking change was made to our node API -- that was our mistake. Discussion is occurring on the kube-register issue.

**Comment by bgrant0607** [comment]
As for putting kube-register in the Kubernetes repo, that would make sense in the context of creating a Fleet cloudprovider: #2890.

**Comment by lavalamp** [comment]
IRL discussion w/ @bgrant0607 summary--

We are going to undo the breaking API change we made.

If we had a fleet cloud provider, then kube-register could conceivably live with that code.

**Comment by AntonioMeireles** [comment]
@lavalamp once the undo is done would be handy to have pushed a new pre-release, as that would allow people on CoreOS/fleetland to do testing with minimal effort...

**Comment by rjnagal** [comment]
we are making a release today.

**Comment by metral** [comment]
im still hitting issues with regards to registering using the latest v0.14.2. v1beta3 looks like its been disabled from the api, so using v1beta2 still gives issues about missing metadata name, even though i supply it.

here's the error:
```
{
  "kind": "Status",
  "creationTimestamp": null,
  "apiVersion": "v1beta2",
  "status": "Failure",
  "message": "Node \"\" is invalid: [metadata.name: required value, spec.ExternalID: required value]",
  "reason": "Invalid",
  "details": {
    "kind": "Node",
    "causes": [
      {
        "reason": "FieldValueRequired",
        "message": "metadata.name: required value",
        "field": "metadata.name"
      },
      {
        "reason": "FieldValueRequired",
        "message": "spec.ExternalID: required value",
        "field": "spec.ExternalID"
      }
    ]
  },
  "code": 422
}
```

not to mention that my apiserver is riddled with issues about not being able to create the master namespace - here's a snippet:

```
Apr 08 07:31:13 kubernetes-master-0 kube-apiserver[1270]: I0408 07:31:13.694830    1270 plugins.go:70] No cloud provider specified.
Apr 08 07:31:13 kubernetes-master-0 kube-apiserver[1270]: [restful] 2015/04/08 07:31:13 log.go:30: [restful/swagger] listing is available at https://10.208.197.175:6443/swaggerapi/
Apr 08 07:31:13 kubernetes-master-0 kube-apiserver[1270]: [restful] 2015/04/08 07:31:13 log.go:30: [restful/swagger] https://10.208.197.175:6443/swaggerui/ is mapped to folder /swagger-ui/
Apr 08 07:31:13 kubernetes-master-0 kube-apiserver[1270]: I0408 07:31:13.719488    1270 server.go:307] Serving read-only insecurely on 10.208.197.175:7080
Apr 08 07:31:13 kubernetes-master-0 kube-apiserver[1270]: I0408 07:31:13.719502    1270 server.go:334] Serving securely on 10.208.197.175:6443
Apr 08 07:31:13 kubernetes-master-0 kube-apiserver[1270]: I0408 07:31:13.719512    1270 server.go:362] Serving insecurely on 0.0.0.0:8080
Apr 08 07:31:13 kubernetes-master-0 kube-apiserver[1270]: E0408 07:31:13.721155    1270 publish.go:63] Can't create master namespace: 502:  (unhandled http status [Temporary Redirect] with body []) [0]
Apr 08 07:31:13 kubernetes-master-0 kube-apiserver[1270]: E0408 07:31:13.721368    1270 publish.go:38] Can't create master namespace: 502:  (unhandled http status [Temporary Redirect] with body []) [0]
Apr 08 07:31:13 kubernetes-master-0 kube-apiserver[1270]: E0408 07:31:13.722404    1270 publish.go:42] Can't create rw service: 502:  (unhandled http status [Temporary Redirect] with body []) [0]
Apr 08 07:31:13 kubernetes-master-0 kube-apiserver[1270]: E0408 07:31:13.722936    1270 publish.go:67] Can't create ro service: 502:  (unhandled http status [Temporary Redirect] with body []) [0]
Apr 08 07:31:13 kubernetes-master-0 kube-apiserver[1270]: I0408 07:31:13.722903    1270 publish.go:153] setting endpoints for master service "kubernetes" to &{{ } {kubernetes  default    0001-01-01 00:00:00 +0000 UTC <nil> map[] map[]} [{
Apr 08 07:31:13 kubernetes-master-0 kube-apiserver[1270]: E0408 07:31:13.724099    1270 publish.go:45] Can't create rw endpoints: 502:  (unhandled http status [Temporary Redirect] with body []) [0]
Apr 08 07:31:13 kubernetes-master-0 kube-apiserver[1270]: I0408 07:31:13.724193    1270 publish.go:153] setting endpoints for master service "kubernetes-ro" to &{{ } {kubernetes-ro  default    0001-01-01 00:00:00 +0000 UTC <nil> map[] map
Apr 08 07:31:13 kubernetes-master-0 kube-apiserver[1270]: E0408 07:31:13.725342    1270 publish.go:70] Can't create ro endpoints: 502:  (unhandled http status [Temporary Redirect] with body []) [0]
Apr 08 07:31:15 kubernetes-master-0 kube-apiserver[1270]: I0408 07:31:15.286716    1270 server.go:344] Using self-signed cert (/var/run/kubernetes/apiserver.crt, /var/run/kubernetes/apiserver.key)
Apr 08 07:31:23 kubernetes-master-0 kube-apiserver[1270]: E0408 07:31:23.725549    1270 publish.go:38] Can't create master namespace: 502:  (unhandled http status [Temporary Redirect] with body []) [0]
Apr 08 07:31:23 kubernetes-master-0 kube-apiserver[1270]: E0408 07:31:23.726792    1270 publish.go:63] Can't create master namespace: 502:  (unhandled http status [Temporary Redirect] with body []) [0]
Apr 08 07:31:23 kubernetes-master-0 kube-apiserver[1270]: E0408 07:31:23.727476    1270 publish.go:42] Can't create rw service: 502:  (unhandled http status [Temporary Redirect] with body []) [0]
```

**Comment by bgrant0607** [comment]
v1beta3 will be enabled by default in the next release: 0.15. You can enable it in 0.14.x by following the instructions here:
https://github.com/GoogleCloudPlatform/kubernetes/tree/master/examples/guestbook/v1beta3

In v1beta1 and v1beta2, you need to provide `id` rather than `metadata.name`. See:
http://kubernetes.io/third_party/swagger-ui/#!/v1beta1/createNode

The error message is using the wrong version to describe the missing field, sorry. (#3084)

**Comment by bgrant0607** [comment]
Once id is set correctly, externalID should be initialized to the id by default.

**Comment by metral** [comment]
@bgrant0607 thanks for the info.

hmm, i switched to Id for v1beta2 per the createNode docs but its still giving me a server error:

```
{
  "kind": "Status",
  "creationTimestamp": null,
  "apiVersion": "v1beta2",
  "status": "Failure",
  "message": "502:  (unhandled http status [Temporary Redirect] with body []) [0]",
  "code": 500
}
<nil>
```

i wonder if its tied to the issues with my apiserver not being able to create the rw services & namespaces from my previous comment

**Comment by bgrant0607** [comment]
My guess is that your apiserver can't talk to etcd. I suggest filing a separate issue for that, or moving the conversation to IRC (#google-containers on freenode). Also, take a look at https://github.com/GoogleCloudPlatform/kubernetes/wiki/Debugging-FAQ

**Comment by metral** [comment]
Just wanted to follow up for anyone else reading, the apiserver errors I was facing were in fact due to it not being able to talk to the etcd server - fixed that.

also, I used Id after the apiserver could talk to the etcdserver and all is good wrt registering nodes in v1beta2, so v0.14.2 registration is fine as far i'm aware. thx for the help

**Comment by bgrant0607** [comment]
@metral Great!

**Comment by pires** [comment]
Can confirm it works, thanks.

**Comment by erimatnor** [comment]
I am experiencing the same issue as metral (problem to connect to etcd). Anyone knows a solutions? (Running v0.14.2)

**Comment by metral** [comment]
@erimatnor i am running a private discovery service w/ etcd v2.0.9: https://gist.github.com/metral/5a7d728e70e1fc9091cc

and since v2.0.x of etcd uses new 2379/2380 ports, i had to reflect that in running master api server (https://gist.github.com/metral/7bf5067799dfccdbd4a2) where the ETCD_SERVERS is the public/private ip of the discovery service

hope that helps
